src/compiler/checker.ts

```diff
@@ -19501,7 +19502,7 @@ namespace ts {
             undefinedDiagnostic?: DiagnosticMessage,
             nullOrUndefinedDiagnostic?: DiagnosticMessage
         ): Type {
-            if (type.flags & TypeFlags.Unknown) {
+            if (strictNullChecks && type.flags & TypeFlags.Unknown) {

```

Comment:
The `strictNullChecks` check should have been here all along, yes?

Reply:
We think so, yeah.